workers/unity/Packages/com.improbable.gdk.querybasedinteresthelper/InterestHelper.cs

```diff
@@ -0,0 +1,220 @@
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core;
+
+namespace Improbable.Gdk.QueryBasedInterest
+{
+    /// <summary>
+    ///     Utility class to help construct Interest component snapshots.
+    /// </summary>
+    public class InterestHelper
+    {
+        private readonly Dictionary<uint, ComponentInterest> interest;
+
+        private InterestHelper(Dictionary<uint, ComponentInterest> interest)
+        {
+            this.interest = interest ?? new Dictionary<uint, ComponentInterest>();
+        }
+
+        /// <summary>
+        ///     Creates a new InterestHelper object.
+        /// </summary>
+        /// <returns>
+        ///     A new InterestHelper object.
+        /// </returns>
+        public static InterestHelper Begin()
+        {
+            return new InterestHelper(null);
+        }
+
+        /// <summary>
+        ///     Creates a new InterestHelper object given an existing Interest component.
+        /// </summary>
+        /// <param name="interest">
+        ///     An existing Interest component.
+        /// </param>
+        /// <returns>
+        ///     An InterestHelper object.
+        /// </returns>
+        public static InterestHelper From(Interest.Component interest)
+        {
+            return new InterestHelper(interest.ComponentInterest
+                .ToDictionary(entry => entry.Key, entry => entry.Value));
+        }
+
+        /// <summary>
+        ///     Creates a new InterestHelper object from the content of an existing Interest component.
+        /// </summary>
+        /// <param name="interest">
+        ///     The underlying dictionary of an Interest component.
+        /// </param>
+        /// <returns>
+        ///     An InterestHelper object.
+        /// </returns>
+        public static InterestHelper From(Dictionary<uint, ComponentInterest> interest)
+        {
+            return new InterestHelper(interest.ToDictionary(entry => entry.Key, entry => entry.Value));
+        }
+
+        /// <summary>
+        ///     Mutates the given interest component.
+        /// </summary>
+        /// <param name="interest">
+        ///     An existing Interest component.
+        /// </param>
+        /// <returns>
+        ///     An InterestHelper object.
+        /// </returns>
+        public static InterestHelper Mutate(Interest.Component interest)
```

Comment:
What does mutate mean for these functions. I'm not sure, looking at the dictionary one especially, I would expect the given dictionary to be mutated or the one I'm calling this from.

Reply:
Begin/From/Mutate are just ways of initialising the helper 

begin -> new underlying dictionary 

from -> deepcopy and modify a copy of the dictionary (and use ToComponentInterest to get the new copy out at the end)

mutate -> use the reference and add/remove/clear queries from the given dictionary directly (so you don't have to return anything at the end)